jetty-util/src/main/java/org/eclipse/jetty/util/component/ContainerLifeCycle.java

```diff
@@ -154,14 +155,23 @@ protected void doStop() throws Exception
         super.doStop();
         List<Bean> reverse = new ArrayList<>(_beans);
         Collections.reverse(reverse);
+        MultiException mex = new MultiException();
         for (Bean b : reverse)
         {
             if (b._managed==Managed.MANAGED && b._bean instanceof LifeCycle)
             {
                 LifeCycle l = (LifeCycle)b._bean;
-                stop(l);
+                try
+                {
+                    stop(l);
+                }
+                catch(Throwable th)
+                {
+                    mex.add(th);
+                }
             }
         }
+        mex.ifExceptionThrow();
```

Comment:
Should not this change also made to `destroy()` ?

Reply:
No. A failed component might be fixed and restarted.  destroy is something that is called just before all references to the object are dropped (or to achieve dropping of all references).

Reply:
@gregw are you sure ? If the first component that is destroyed throws an exception, all the others will not be destroyed.

Reply:
this is in doStop, so nothing to do with destroy.  However your point you raise about exceptions from destroy is a good one at it should catch throwable and keep destroying....

Reply:
@sbordet I modified destroy